app/react/Library/components/PermissionsFilter.tsx

```diff
@@ -1,66 +1,145 @@
-import React from 'react';
-import SelectFilter from 'app/Library/components/SelectFilter';
+import React, { useMemo } from 'react';
 import FormGroup from 'app/DocumentForm/components/FormGroup';
+import { MultiSelect } from 'app/Forms';
 import { Aggregations } from 'shared/types/aggregations';
 import { NeedAuthorization } from 'app/Auth';
 import { Translate } from 'app/I18N';
-import { Icon } from 'app/UI';
+import { Control } from 'react-redux-form';
+import { connect } from 'react-redux';
+import { UserSchema } from 'shared/types/userType';
+import { IImmutable } from 'shared/types/Immutable';
+import Immutable from 'immutable';
+
+type PermissionLevel = 'read' | 'write';
+
+interface PermissionsValue {
+  level: PermissionLevel;
+  refId: string;
+}
 
 export interface PermissionsFilterProps {
-  onChange: () => void;
+  onChange: (values: PermissionsValue[]) => void;
   aggregations: Aggregations;
 }
 
 const filteredAggregation = (aggregations: Aggregations, key: string) => {
-  const bucket = (aggregations?.all?.permissions?.buckets || []).find(a => a.key === key) || {
+  const bucket = (aggregations?.all?.['_permissions.self']?.buckets || []).find(
+    a => a.key === key
+  ) || {
     filtered: { doc_count: 0 },
   };
   return bucket.filtered.doc_count;
 };
 
 const generateOptions = (aggregations: Aggregations) => [
   {
-    label: (
-      <>
-        <Icon icon="pencil-alt" />
-        &nbsp;
-        <Translate>Can edit</Translate>
-      </>
-    ),
+    label: <Translate>Can edit</Translate>,
     title: 'Can edit',
     value: 'write',
+    icon: { type: 'Icons', _id: 'pencil-alt' },
     results: filteredAggregation(aggregations, 'write'),
   },
   {
-    label: (
-      <>
-        <Icon icon="file" />
-        &nbsp;
-        <Translate>Can view</Translate>
-      </>
-    ),
+    label: <Translate>Can view</Translate>,
     title: 'Can view',
     value: 'read',
+    icon: { type: 'Icons', _id: 'file' },
     results: filteredAggregation(aggregations, 'read'),
   },
 ];
 
+export const PermissionsFilterUncontrolled = connect(
+  ({ user }: { user: IImmutable<UserSchema> }) => ({
+    user,
+  })
+)(
+  ({
+    value = Immutable.fromJS([]),
+    onChange,
+    aggregations,
+    user,
+  }: PermissionsFilterProps & { user: IImmutable<UserSchema>; value: PermissionsValue[] }) => {
+    const refIds: string[] = useMemo(
+      () => [user.get('_id') as string, ...(user.toJS().groups?.map(g => g._id as string) || [])],
+      [user]
+    );
+
+    const onChangeHandler = (newValues: PermissionLevel[]) => {
+      onChange(
+        newValues.reduce(
+          (filters: PermissionsValue[], level: PermissionLevel) =>
+            filters.concat(
+              refIds.map(refId => ({
+                refId,
+                level,
+              }))
+            ),
+          []
+        )
+      );
+    };
+
+    const mappedValue = useMemo(
+      () =>
+        ['read', 'write'].filter(level =>
```

Comment:
Maybe connecting this to the type, somehow like in this [SO thread](https://stackoverflow.com/questions/44497388/typescript-array-to-string-literal-type)? I've never tried it, so this is more just a discussive idea.

Reply:
It makes sense. I didn't do it because I thought it was an overkill (since it's not more type safe that currently) but now you suggested it: there I go :)